### PR TITLE
fix: wrong json property name for difficulty rating

### PR DESCRIPTION
**Pull request by matte-ek**
Fixes some minor stuff I caught while trying out the library today.

*  [Rename DifficultyRating to StarRating](https://github.com/minisbett/osu-sharp/commit/f0174e63d56bdd691f427e06e6fc21b33f9d8cd7) as per the [osu! documentation](https://osu.ppy.sh/docs/index.html#beatmapdifficultyattributes:~:text=of%20length%20100.-,BeatmapDifficultyAttributes,-Represent%20beatmap%20difficulty)
* ~~[Add description attributes to RankedStatus enum](https://github.com/minisbett/osu-sharp/commit/6455fda6590bc45757f2a34a596c0a3e1609dfcc) since otherwise the `GetBeatmapAsync` request would cause an exception~~

I was going to add the ability to pass mods onto the `GetDifficultyAttributesAsync` endpoint, but it seems like the API expects the data in the body of the request as JSON, and I wasn't sure how you would like to deal with cases like these, [my changes](https://github.com/matte-ek/osu-sharp/commit/f4ecaa63e15afe35922651b1a05013d8d35979bb) FWIW.

Nice work on the library so far though, nice and easy to work with, and code is very nice.

### CODE REVIEW

**Comment by minisbett** [comment]
Thanks!

Yeah those two are oversights. Although the `RankedStatus` has already been fixed, the star rating field name was definitely an oversight. As for passing mods to `GetDifficultyAttributesAsync`, I can add that, sure.

File: OsuSharp/Models/Beatmaps/DifficultyAttributes.cs [hunk 1]
```diff
@@ -19,8 +19,8 @@ public class DifficultyAttributes
   /// <summary>
   /// The difficulty rating of the beatmap.
   /// </summary>
-  [JsonProperty("difficulty_rating")]
-  public float DifficultyRating { get; private set; }
+  [JsonProperty("star_rating")]
+  public float StarRating { get; private set; }
```
Comment (minisbett) [inline OsuSharp/Models/Beatmaps/DifficultyAttributes.cs]:
```suggestion
  public float DifficultyRating { get; private set; }
```

**Review by minisbett — CHANGES_REQUESTED** [review]

**Comment by matte-ek** [comment]
Oh I see, didn't see that commit. Dropped the commit from my branch in case you want to merge this one.

**Comment by minisbett** [comment]
If you'd be interested in contributing by adding mod-support for the difficulty attributes:

The way I'd imagine an implementation would simply having 3 different overloads for specifying either a string (eg. `HDDT`), an array, or the bitwise integer flag. It seems like I also didn't implement the `ruleset` parameter, which could be implemented there. (`ruleset_id` can be ignored)

As for the fact it is a post request, I currently have my main request handler `GetFromJsonAsync`, which I manually pass a `HttpMethod.Post` to. A dirty hack, I know. Maybe it could be split up into a `GetAsync` and `PostAsync`, which both have their own specific code but call a private method inside `OsuApiClient` (eg. `SendAsync`) which then handles all the other request-related code. The `PostAsync` method could then allow for specifying the body, which is required to pass mods & co.


Edit: I checked your changes you linked, and I think they're pretty good. Feel free to PR them so we can work from there!

**Pull request merged**